src/org/opensolaris/opengrok/history/FileHistoryCache.java

```diff
@@ -309,9 +302,35 @@ private void storeFile(History histNew, File file, Repository repo) throws Histo
                 }
             }
         } catch (IOException ex) {
-            // Ideally we would want to catch the case when incremental update
-            // is done but the cached file is not there however we do not have
-            // the data to do it here.
+            LOGGER.log(Level.SEVERE, "Cannot open history cache file " + cacheFile.getPath());
+        }
+
+        return history;
+    }
+
+    /**
+     * Store history object (encoded as XML and compressed with gzip) in a file.
+     *
+     * @param history history object to store
+     * @param file file to store the history object into
+     * @param repo repository for the file
+     * @param mergeHistory whether to merge the history with existing or store the histNew as is
+     * @throws HistoryException
+     */
+    private void storeFile(History histNew, File file, Repository repo,
+            boolean mergeHistory) throws HistoryException {
+
+        File cacheFile = getCachedFile(file);
+        History history = histNew;
+
+        File dir = cacheFile.getParentFile();
+        if (!dir.isDirectory() && !dir.mkdirs()) {
+            throw new HistoryException(
+                    "Unable to create cache directory '" + dir + "'.");
+        }
+
+        if (mergeHistory && cacheFile.exists()) {
+            history = mergeOldAndNewHistory(cacheFile, histNew, repo);
         }
 
         writeHistoryToFile(dir, history, cacheFile);
```

Comment:
What happens here when mergeOldAndNewHistory() returns null. Looks like it will fail in writeHistoryToFile() later.